truffle/src/com.oracle.truffle.polyglot/src/com/oracle/truffle/polyglot/HostMethodDesc.java

```diff
@@ -260,7 +260,7 @@ public final Object invoke(Object receiver, Object[] arguments) throws Throwable
                 try {
                     return invokeHandle(methodHandle, receiver, arguments);
                 } catch (ClassCastException ex) {
-                    throw UnsupportedTypeException.create(arguments);
+                    throw UnsupportedTypeException.create(arguments, ex.getMessage());
```

Comment:
it took me a lot of time to figure out what's going with an UnsupportedTypeException which is not obvious.

for me i got error msg:
```
PojoValidationException while performing object save: org.graalvm.polyglot.PolyglotException: TypeError: invokeMember (myMethod) on JavaObject[com.example.ServiceImpl@4ba5c0f0 (com.sun.proxy.$Proxy167)] failed due to: UnsupportedTypeException:
```

which actually error of `java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long` when i debug in depth.